### PR TITLE
mod_wires: Remove 2 char limit on typeselect search

### LINKED ISSUE #3192: Typeselect action does not reset results when going from 3 to 2 characters
## Steps to reproduce

Setup the type select or use the media select in the admin

## Current behavior

Enter at least 3 characters in the input box, wait for results to appear, and remove a character. The previously found results are still visible.

## Expected behavior

No results should be available when less than 3 characters should be reset. When the user previously had 3 characters in the input field

## Thoughts

Why is the query only run after at least 3 characters are entered. This is behavior is probably easier to get right when the typeselect is always run when something is typed in the input box.

### PR DESCRIPTION
**Pull request by mmzeeman**
### Description

Fix #3192 3192

Removes the two character limit on type select searches. As a result, previously found resources are removed when there is no input.

### Checklist

- [x] documentation updated
- [ ] tests added
- [x] no BC breaks

### CODE REVIEW

**Review by mworrell — APPROVED** [review]

File: apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js [hunk 1]
```diff
@@ -893,13 +893,9 @@ function z_typeselect(ElementId, postbackInfo)
 
     z_input_updater = setTimeout(function()
     {
-        var obj = $('#'+ElementId);
-
-        if(obj.val().length >= 2)
```
Comment (mworrell) [inline apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: Maybe make it >= 1 ?

Then we don't post on 0 chars.

**Review by mworrell — APPROVED** [review]

File: apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js (hunk 1 above)
Comment (mmzeeman) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: But then it won't reset the search results when you go from 1 to 0, which is what users expect.

Comment (mworrell) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: Should we return the empty result for the empty query?  Or just anything that matches?

Comment (mmzeeman) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: The example code in the documentation is also interesting. A commented out wire statement.

Screen Shot 2022-11-28 at 12 33 03

Comment (mmzeeman) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: Side issue.. the color scheme for example code on zotonic.com makes it look like wire statements are comments.

Screen Shot 2022-11-28 at 12 38 03

Comment (mmzeeman) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: Ah.. `%`

Comment (mmzeeman) [reply apps/zotonic_mod_wires/priv/lib/js/apps/zotonic-wired.js:898]: The typeaction uses the autocomplete query. This returns 0 result without querying the database. 

I've updated the documentation to correctly use an input type="search" input box. This makes sure a "Search" button appears on the keyboard of an iOS device. On Safari an Chrome the search box can also be cleared by typing <kbd>Esc</kbd> or clicking on a small cross appearing at the right of the input box. For the clickable cross the type of the wire must be changed from `keyup` to `input`.

**Pull request merged**